queries/words.py

```diff
@@ -0,0 +1,238 @@
+    r"^hvernig skrifar maður (?:orðið|nafnið) (.+)$",
+    r"^hvernig skrifar maður (.+)$",
+    r"^hvernig stafar maður (?:orðið|nafnið) (.+)$",
+    r"^hvernig stafar maður (.+)$",
+    r"^hvernig er (?:orðið|nafnið) (.+) stafsett$",
+    r"^hvernig er (.+) stafsett$",
+    r"^hvernig er (?:orðið|nafnið) (.+) skrifað$",
+    r"^hvernig er (.+) skrifað$",
+    r"^hvernig er (?:orðið|nafnið) (.+) stafað$",
+    r"^hvernig er (.+) stafað$",
+    r"^hvernig skal stafa (.+)$",
+    r"^hvernig skal stafa (?:orðið|nafnið) (.+)$",
+    r"^hvernig stafast (?:orðið|nafnið) (.+)$",
+)
+
+# Add "nafnið" non-capture groups
+
+_DECLENSION_RX = (
+    r"^hvernig beygi ég (?:orðið|nafnið) (.+)$",
+    r"^hvernig beygirðu (?:orðið|nafnið) (.+)$",
+    r"^hvernig á að beygja (?:orðið|nafnið) (.+)$",
+    r"^hvernig á ég að beygja (?:orðið|nafnið) (.+)$",
+    r"^hvernig á maður að beygja (?:orðið|nafnið) (.+)$",
+    r"^hvernig beygir maður (?:orðið|nafnið) (.+)$",
+    r"^hvernig beygir maður (?:orðið|nafnið) (.+)$",
+    r"^hvernig beygist (?:orðið|nafnið) (.+)$",
+    r"^hvernig skal beygja (?:orðið|nafnið) (.+)$",
+    r"^hvernig er (?:orðið|nafnið) (.+) beygt$",
+    r"^hverjar eru beygingarmyndir (?:orðsins|nafnsins) (.+)$",
+    r"^hvað eru beygingarmyndir (?:orðsins|nafnsins) (.+)$",
+)
+
+
+def lookup_best_word(word):
+    """ Look up word in BÍN, pick right one acc. to a criterion. """
+    res = BIN_Db().lookup_nominative(word)
+    if not res:
+        # Try with uppercase first char
+        res = BIN_Db().lookup_nominative(word.capitalize())
+        if not res:
+            return None
+
+    # OK, we have one or more matching words
+    if len(res) == 1:
+        m = res[0]
+    else:
+        # TODO: Pick best result (prefer nouns vs. adjectives, etc?)
+        m = res[0]  # For now
+
+    # TODO: Fix this. If more than one declesion possible, list them all
+    def sort_by_preference(m_list):
+        """ Discourage rarer declension forms, i.e. ÞGF2 and ÞGF3 """
+        return sorted(m_list, key=lambda m: "2" in m.beyging or "3" in m.beyging)
+
+    # Look up all cases of the word in BÍN
+    with BIN_Db().get_db() as db:
+        nom = m.stofn
+        acc = db.cast_to_accusative(nom, meaning_filter_func=sort_by_preference)
+        dat = db.cast_to_dative(nom, meaning_filter_func=sort_by_preference)
+        gen = db.cast_to_genitive(nom, meaning_filter_func=sort_by_preference)
+        return nom, acc, dat, gen
+
+
+_NOT_IN_BIN_MSG = "Orðið '{0}' fannst ekki í Beyingarlýsingu íslensks nútímamáls."
+
+
+def declension_answer_for_word(word, query):
+    """ Look up all morphological forms of a given word,
+        construct natural language response. """
+
+    # Look up in BÍN
+    forms = lookup_best_word(word)
+
+    if not forms:
+        return gen_answer(_NOT_IN_BIN_MSG.format(word))
+
+    answ = ", ".join(forms)
+    response = dict(answer=answ)
+    cases_desc = "Hér er {0}, um {1}, frá {2}, til {3}".format(*forms)
```

Comment:
This only works for nouns, right? So lookup_best_word should filter for nouns only. Also, how should the definitive article (greinir) be handled? If I ask about "kötturinn", do I get the answers with the definitive article?